presto-main/src/main/java/io/prestosql/metadata/FunctionManager.java

```diff
@@ -109,4 +109,9 @@ public FunctionHandle lookupInternalCastFunction(String name, TypeSignature from
     {
         return functionRegistry.lookupInternalCastFunction(name, fromType, toType);
     }
+
+    public FunctionMetadata getFunctionMetadata(FunctionHandle functionHandle)
```

Comment:
Why this is `getFunctionMatadata(FunctionHandle)` rather than `FunctionHandle.getFunctionMetadata`? Also, conceptually, it seems to me that you'd only need `FunctionMetadata` at plan time, so in some ways `FunctionMetadata` can be `FunctionHandle` (or the other way around). Why do we need both?